test/test_cartesian_product.cpp

```diff
@@ -110,6 +111,167 @@ constexpr bool test_cartesian_product()
+        static_assert(std::same_as<flux::rvalue_element_t<C>, std::tuple<int&&, float&&, unsigned long long&&>>);
+
+        static_assert(std::same_as<flux::element_t<C const>, std::tuple<int&, float&, unsigned long long&>>);
+        static_assert(std::same_as<flux::value_t<C const>, std::tuple<int, float, unsigned long long>>);
+        static_assert(std::same_as<flux::rvalue_element_t<C const>, std::tuple<int&&, float&&, unsigned long long&&>>);
+
+        STATIC_CHECK(flux::size(cart) == 2 * 4 * 3);
+
+        STATIC_CHECK(check_equal(cart, {
+            std::tuple{100, 1.0f, 0ULL},
+            std::tuple{100, 1.0f, 2ULL},
+            std::tuple{100, 1.0f, 4ULL},
+            std::tuple{100, 2.0f, 0ULL},
+            std::tuple{100, 2.0f, 2ULL},
+            std::tuple{100, 2.0f, 4ULL},
+            std::tuple{100, 3.0f, 0ULL},
+            std::tuple{100, 3.0f, 2ULL},
+            std::tuple{100, 3.0f, 4ULL},
+            std::tuple{100, 4.0f, 0ULL},
+            std::tuple{100, 4.0f, 2ULL},
+            std::tuple{100, 4.0f, 4ULL},
+            std::tuple{200, 1.0f, 0ULL},
+            std::tuple{200, 1.0f, 2ULL},
+            std::tuple{200, 1.0f, 4ULL},
+            std::tuple{200, 2.0f, 0ULL},
+            std::tuple{200, 2.0f, 2ULL},
+            std::tuple{200, 2.0f, 4ULL},
+            std::tuple{200, 3.0f, 0ULL},
+            std::tuple{200, 3.0f, 2ULL},
+            std::tuple{200, 3.0f, 4ULL},
+            std::tuple{200, 4.0f, 0ULL},
+            std::tuple{200, 4.0f, 2ULL},
+            std::tuple{200, 4.0f, 4ULL}
+        }));
+
+        STATIC_CHECK(flux::distance(cart, cart.first(), cart.last()) == 2 * 4 * 3);
+
+        {
+            auto cur = flux::next(cart, cart.first(), 3);
+            STATIC_CHECK(cart[cur] == std::tuple{100, 2.0f, 0ULL});
+            flux::inc(cart, cur, -3);
+            STATIC_CHECK(cart[cur] == std::tuple{100, 1.0f, 0ULL});
+        }
+    }
+
+    {
+        auto cart = flux::cartesian_product(flux::ints(0, 4), flux::ints(0, 2), flux::ints(0, 3));
+
+        using C = decltype(cart);
+
+        static_assert(flux::sequence<C>);
+        static_assert(flux::multipass_sequence<C>);
+        static_assert(flux::bidirectional_sequence<C>);
+        static_assert(flux::random_access_sequence<C>);
+        static_assert(not flux::contiguous_sequence<C>);
+        static_assert(flux::bounded_sequence<C>);
+        static_assert(flux::sized_sequence<C>);
+
+        static_assert(flux::sequence<C const>);
+        static_assert(flux::multipass_sequence<C const>);
+        static_assert(flux::bidirectional_sequence<C const>);
+        static_assert(flux::random_access_sequence<C const>);
+        static_assert(not flux::contiguous_sequence<C const>);
+        static_assert(flux::bounded_sequence<C const>);
+        static_assert(flux::sized_sequence<C const>);
+
+        static_assert(std::same_as<flux::element_t<C>, std::tuple<long, long, long>>);
```

Comment:
`C` is `cartesian_product(ints(), ints(), ints())`, so its element type is `std::tuple<distance_t, distance_t, distance_t>`, where `distance_t` is our alias for `ptrdiff_t`. On 32-bit Windows `ptrdiff_t` is just `int`, hence the Windows static assert failures. (But it would fail on 32-bit Linux too, if tested that.)

One possible solution:

```suggestion
        static_assert(std::same_as<flux::element_t<C>, std::tuple<flux::distance_t, flux::distance_t, flux::distance_t>>);
```

(And the same for the other static asserts)

Reply:
Fixed in 62a25c31de697cff4fa163e48eda1409a3c81059